src/langauge-servers/langaugeServer.ts

```diff
@@ -0,0 +1,119 @@
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+//import * as path from 'path';
+import * as rpc from "vscode-ws-jsonrpc";
+import { RequestMessage } from "vscode-ws-jsonrpc";
+import * as bridgeServer from "vscode-ws-jsonrpc/lib/server";
+import * as lsp from "vscode-languageserver";
+import * as cmd from 'child_process'
+
+console.log("Starting langserver!!!!")
+
+export interface langServerStartConfig {
+    LangaugeName: string,
+    langserverCommand: string,
+    langserverCommandArgs: string[]
+}
+
+// The init request from the frontend indicates on which repo should the langserver run:
+// the uri is sent like this: "file:///<repo_id>", repo_id is the id in repoStore.
+// the repo's fullPath is the one sent to the langserver itself.
+
+// Every other request to the langserver sends which file is it asking about,
+// the file's uri is sent like this "file://<relative_path_to_file>",
+// the langserver uses fullPath so we use the repoStore to get it.
+export const launchLangaugeServer = (socket: rpc.IWebSocket, startConfig: langServerStartConfig) => {
+
+    // Since we can't know which response is a response to definition,
+    // we save up all the messages ids for definitions requests.
+    // Since in lsp, a request message and its response has the same message id.
+    const definitionsIds = new Set();
+
+    const reader = new rpc.WebSocketMessageReader(socket);
+    const writer = new rpc.WebSocketMessageWriter(socket);
+    let repo: Repo = null;
+
+    const langserverProcessName = 'Rookout-' + startConfig.LangaugeName + '-LangServer'
+    const serverConnection = bridgeServer.createServerProcess(langserverProcessName, startConfig.langserverCommand, startConfig.langserverCommandArgs)
+    const socketConnection = bridgeServer.createConnection(reader, writer, () => {socket.dispose(); serverConnection.dispose()});
+    
+
+    bridgeServer.forward(socketConnection, serverConnection, message => {
+        if (rpc.isRequestMessage(message)) {
+            if (message.method === lsp.InitializeRequest.type.method) {
+                const initializeParams = message.params as lsp.InitializeParams;
+                initializeParams.processId = process.pid;
+                
+                const repoId = initializeParams.workspaceFolders[0].uri.replace('file:///', '')
+                repo = repStore.getRepoById(repoId)
+
+                // rootUri and rootPath are considered deprecated by the vscode's lsp and they are the only way to indicate 
+                // to the language server the workspace folder
+                initializeParams.rootUri = initializeParams.workspaceFolders[0].uri = initializeParams.workspaceFolders[0].name = 'file://' + repo.fullpath
+                initializeParams.rootPath = repo.fullpath
```

Comment:
they reference to `workspaceFolders` isn't it covering what we need?

Reply:
I should but it doesn't, probably because of the lsp version we use, but that is fine, the comment was written because that `rootUri` and `rootPath` are marked by vscode as deprecated